src/sensors.cpp

```diff
@@ -1,261 +1,201 @@
+        output_total = output_total - output_avgs[avg_out_idx];
+        output_avgs[avg_out_idx] = pulses;
+        output_total = output_total + pulses;
         avg_out_idx = (avg_out_idx+1)%RPM_SAMPLES_DEBOUNCE;
+    } else {
+
     }
+    return true;
 }
 
-esp_err_t Sensors::init_sensors(){
-    const pcnt_config_t pcnt_cfg_n2{
-        .pulse_gpio_num = pcb_gpio_matrix->n2_pin,
-        .ctrl_gpio_num = PCNT_PIN_NOT_USED,
-        .lctrl_mode = PCNT_MODE_KEEP,
-        .hctrl_mode = PCNT_MODE_KEEP,
-        .pos_mode = PCNT_COUNT_INC,
-        .neg_mode = PCNT_COUNT_DIS,
-        .counter_h_lim = PCNT_H_LIM,
-        .counter_l_lim = 0,
-        .unit = PCNT_N2_RPM,
-        .channel = PCNT_CHANNEL_0};
-
-    const pcnt_config_t pcnt_cfg_n3{
-        .pulse_gpio_num = pcb_gpio_matrix->n3_pin,
-        .ctrl_gpio_num = PCNT_PIN_NOT_USED,
-        .lctrl_mode = PCNT_MODE_KEEP,
-        .hctrl_mode = PCNT_MODE_KEEP,
-        .pos_mode = PCNT_COUNT_INC,
-        .neg_mode = PCNT_COUNT_DIS,
-        .counter_h_lim = PCNT_H_LIM,
-        .counter_l_lim = 0,
-        .unit = PCNT_N3_RPM,
-        .channel = PCNT_CHANNEL_0};
-
+const pcnt_glitch_filter_config_t glitch_filter = {
+    .max_glitch_ns = 1000
+};
+
+esp_err_t configure_pcnt(const char* name, gpio_num_t gpio, pcnt_unit_handle_t* UNIT_HANDLE, pcnt_channel_handle_t* CHANNEL_HANDLE) {
+    ESP_RETURN_ON_ERROR(gpio_set_direction(gpio, GPIO_MODE_INPUT), "SENSORS", "Failed to set %s Pin to Input", name);
+    ESP_RETURN_ON_ERROR(gpio_set_pull_mode(gpio, GPIO_PULLUP_ONLY), "SENSORS", "Failed to set %s Pin to pullup", name);
+    ESP_RETURN_ON_ERROR(pcnt_new_unit(&RPM_UNIT_CFG, UNIT_HANDLE), "SENSORS", "Failed to setup %s RPM PCNT Unit", name);
+    const pcnt_chan_config_t rpm_chan_config = {
+        .edge_gpio_num = gpio,
+        .level_gpio_num = -1,
+        .flags {
+            .invert_edge_input = 0,
+            .invert_level_input = 0,
+            .virt_edge_io_level = 0,
+            .virt_level_io_level = 0,
+            .io_loop_back = 0,
+        }     
+    };
+    ESP_RETURN_ON_ERROR(pcnt_new_channel(*UNIT_HANDLE, &rpm_chan_config, CHANNEL_HANDLE), "SENSORS", "Failed to setup %s RPM PCNT Channel", name);
+    ESP_RETURN_ON_ERROR(
+        pcnt_channel_set_edge_action(*CHANNEL_HANDLE, pcnt_channel_edge_action_t::PCNT_CHANNEL_EDGE_ACTION_INCREASE, pcnt_channel_edge_action_t::PCNT_CHANNEL_EDGE_ACTION_INCREASE),
+        "SENSORS",
+        "Failed to set PCNT actions for %s PCNT",
+        name
+    );
+    ESP_RETURN_ON_ERROR(pcnt_unit_set_glitch_filter(*UNIT_HANDLE, &glitch_filter), "SENSORS", "Failed to set glitch filter for PCNT unit %s", name);
+    ESP_RETURN_ON_ERROR(pcnt_unit_enable(*UNIT_HANDLE), "SENSORS", "Failed to enable PCNT unit %s", name);
+    ESP_RETURN_ON_ERROR(pcnt_unit_start(*UNIT_HANDLE), "SENSORS", "Failed to start PCNT unit %s", name);
+    return ESP_OK;
+}
 
+esp_err_t Sensors::init_sensors(){
```

Comment:
from MISRA: init_sensors() should be init_sensors(void)

Reply:
Fixed in https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gptimer.html